File: 2324-find-triangular-sum-of-an-array/find-triangular-sum-of-an-array.cpp
class Solution {
public:
    int triangularSum(vector<int>& nums) {
        
        while (nums.size() > 1) {
            vector<int> next;

            for (int i = 0; i < nums.size() - 1; i++) {
                next.push_back((nums[i] + nums[i+1]) % 10);
            }

            nums = next;
        }

        return nums[0];
    }
};

File: 2324-find-triangular-sum-of-an-array/find-triangular-sum-of-an-array.java
class Solution {
    public int triangularSum(int[] nums) {
         while(true){
            if(nums.length==1)
                return nums[0]%10;
                int[] sumdp= new int[nums.length-1];
                for(int j=0;j<nums.length-1;j++){
                    sumdp[j]=(nums[j]+nums[j+1])%10;
                }
                if(sumdp.length==1){
                return sumdp[0];
                }

                nums=sumdp;
            }
            // return -1;
        }
    }
